### PR TITLE
[node] mark POSIX-only functions as optional

### PR DESCRIPTION
**Pull request by Josh-Cena**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: See https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/60494
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by amcasey — COMMENTED** [review]
This looks correct - I'm on Windows and I don't have those functions, but it also seems likely to break most consumers (who, presumably, already know if they're completely broken at runtime).  @mcollina might have thoughts?

**Comment by SimonSchick** [comment]
We can merge this into the v18 typings PR I created.

**Comment by amcasey** [comment]
@SimonSchick works for me.  @Josh-Cena ?

**Comment by Josh-Cena** [comment]
Yes, I'm good with making this into v18. Do I need to do anything on my end?

File: types/node/test/process.ts [hunk 1]
```diff
@@ -129,3 +129,26 @@ process.env.TZ = 'test';
 {
     const arch: NodeJS.Architecture = process.arch;
 }
+
+{
+    // $ExpectError
+    process.getgid();
```
Comment (LinusU) [inline types/node/test/process.ts:135]: Potentially we could also test to call all the functions using optional chaining operator, to make sure that the functions still exists and are typed properly?

```suggestion
    // $ExpectError
    process.getgid();
    process.getgid?.();
```

**Review by LinusU — APPROVED** [review]
Neat 👍

**Review by mcollina — APPROVED** [review]
lgtm

File: types/node/test/process.ts (hunk 1 above)
Comment (Josh-Cena) [reply types/node/test/process.ts:135]: Added a bunch more tests

**Comment by peterblazejewicz** [comment]
@SimonSchick should it be merged into current master, or will be handled with cherry-pick/retarget of the branch? thx!

**Comment by rbuckton** [comment]
@SimonSchick, @peterblazejewicz: Is this change being merged into a different PR? Should this PR be closed?

**Comment by peterblazejewicz** [comment]
I just wanted to avoid problems with other work on v18. But this can be merged, I can help with cherry--picking changes into other pr, if requested. thx!
I'll mark as LGTM

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by Josh-Cena** [comment]
Ready to merge

**Pull request merged**